### PR TITLE
Test: Price Floors imp level floorMin

### PR DESCRIPTION
**Pull request by marki1an**
Test for: https://github.com/prebid/prebid-server-java/pull/1975

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 1]
```diff
@@ -386,10 +387,43 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
...
+                floorMin = randomFloorValue
+                floorValue = FLOOR_MIN
+            }
+        }
+
+        and: "Set Floors Provider response with a floorMin"
+        def floorsResponse = PriceFloorData.priceFloorData.tap {
+            modelGroups[0].values = [(rule): FLOOR_MIN]
+        }
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
```
Comment (SerhiiNahornyi) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:413]: For a first request, we don't have floors data we have only a request to start fetching provider values
1) You can pass rules in request to use them from first call
2) The best case request.floorMin =2USD, imp.floorMin =4USD, modelGroup = 3USD (Resolved 4 for bidder)

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 2]
```diff
@@ -341,4 +343,106 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
...
+            ext.prebid.floors.floorMinCur == JPY
+            ext.prebid.floors.floorMin == FLOOR_MIN
+        }
+    }
+
+     def "PBS should choose floorMin from imp[0].ext.prebid.floors when imp[0].ext.prebid.floors is present"() {
+        given: "Default BidRequest with floorMin, floorMinCur"
+        def impExtPrebidFloorMin = BigDecimal.valueOf(4)
+        def bidRequest = bidRequestWithFloors.tap {
+                ext.prebid.floors.floorMin = BigDecimal.valueOf(2)
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:402]: Same. Please don't hardcode magic numbers.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 3]
```diff
@@ -341,4 +343,106 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
...
+                ext.prebid.floors.data.modelGroups[0].values = [(rule): BigDecimal.valueOf(3)]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:403]: Same. Please don't hardcode magic numbers.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 4]
```diff
@@ -341,4 +343,106 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
...
+            imp[0].ext.prebid.floors.floorMinCur == EUR
+            imp[0].ext.prebid.floors.floorMin == FLOOR_MIN
+            ext.prebid.floors.floorMinCur == JPY
+            ext.prebid.floors.floorMin == FLOOR_MIN
+        }
+    }
+
+     def "PBS should choose floorMin from imp[0].ext.prebid.floors when imp[0].ext.prebid.floors is present"() {
+        given: "Default BidRequest with floorMin, floorMinCur"
+        def impExtPrebidFloorMin = BigDecimal.valueOf(4)
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:400]: Please don't hardcode magic numbers.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 5]
```diff
@@ -341,4 +343,106 @@ class PriceFloorsCurrencySpec extends PriceFloorsBaseSpec {
...
+            imp[0].ext.prebid.floors.floorMin == impExtPrebidFloorMin
+            imp[0].ext.prebid.floors.floorValue == impExtPrebidFloorMin
+            imp[0].bidFloor == impExtPrebidFloorMin
+            imp[0].bidFloorCur == USD
+        }
+    }
+
+    def "PBS should choose floorMin from ext.prebid.floors when imp[0].ext.prebid.floor.floorMin is absent"() {
+        given: "Default BidRequest with floorMin"
+        def extPrebidFloorMin = BigDecimal.valueOf(5)
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:426]: Same. Please don't hardcode magic numbers.

**Review by Net-burst — CHANGES_REQUESTED** [review]

**Review by Net-burst — APPROVED** [review]
LGTM

**Pull request merged**